src/components/SignUpModal/SignUpModal.jsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState, useEffect, useId } from "react";
+import PopupWithForm from "../PopupWithForm/PopupWithForm";
+import RegistrationSuccessModal from "../../components/RegistrationSuccessModal/RegistrationSuccessModal";
+
+function SignUpModal({ isOpen, onClose, onOverlayClick, onSignIn }) {
+  // Generate unique IDs for each instance
+  const emailId = useId();
+  const usernameId = useId();
+  const passwordId = useId();
+
+  const [email, setEmail] = useState("");
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [usernameError, setUsernameError] = useState("");
+  const [isFormValid, setIsFormValid] = useState(false);
+  const [showSuccessModal, setShowSuccessModal] = useState(false);
+
+  const validateForm = () => {
+    const isUsernameValid = username.length > 0 && !usernameError;
+    const isPasswordValid = password.length >= 6;
+    setIsFormValid(isUsernameValid && isPasswordValid && email.includes("@"));
+  };
+
+  const handleEmailChange = (e) => {
+    setEmail(e.target.value);
+    validateForm();
+  };
+
+  const handleUsernameChange = (e) => {
+    const value = e.target.value;
+    setUsername(value);
+    if (["existingUser", "takenUsername"].includes(value)) {
+      setUsernameError("Username isn't available");
+    } else {
+      setUsernameError("");
+    }
+    validateForm();
+  };
+
+  const handlePasswordChange = (e) => {
+    setPassword(e.target.value);
+    validateForm();
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (isFormValid) {
+      localStorage.setItem("username", username);
+      localStorage.setItem("email", email);
+      localStorage.setItem("isLoggedIn", "true");
+
+      onSignIn();
+
+      setShowSuccessModal(true);
+      setTimeout(() => {
+        onClose();
+      }, 200);
+    }
+  };
+
+  useEffect(() => {
+    validateForm();
+  }, [email, username, password]);
+
+  return (
+    <>
```

Comment:
`<>` empty tags are not needed if you have a common tag that wraps the whole markup of the component